python/orca/src/bigdl/orca/learn/utils.py

```diff
@@ -94,7 +94,7 @@ def group_index(iter):
     def transform_predict(predictions):
         # list of np array
         if isinstance(predictions[0], list):
-            predictions = np.array(predictions).T.tolist()
+            predictions = np.array(predictions, dtype=object).T.tolist()
```

Comment:
np 1.24 requires `array` to have homogenous shape

Reply:
not correct to be here; we can't treat everything to be object.

Reply:
python 3.7 test already gives the warning:
```python
/home/yinchen/BigDL/python/orca/src/bigdl/orca/learn/utils.py:97: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  predictions = np.array(predictions).T.tolist()
```
Command
```bash
cd python/orca
python -m pytest --show-capture=all --full-trace -r P -v test/bigdl/orca/learn/test_utils.py::TestUtil::test_convert_predict_rdd_to_xshard_multi_output
```

Reply:
@sgwhat Fix this issue?
```
predictions = np.array(predictions).T.tolist()
```